libs/utils/gtest/src/PropertiesEncodingTestSuite.cc

```diff
@@ -1103,4 +1103,29 @@ TEST_F(PropertiesSerializationTestSuite, JsonErrorToCelixStatusTest) {
     EXPECT_EQ(CELIX_ILLEGAL_ARGUMENT, celix_properties_jsonErrorToStatus(json_error_numeric_overflow));
     EXPECT_EQ(CELIX_ILLEGAL_ARGUMENT, celix_properties_jsonErrorToStatus(json_error_item_not_found));
     EXPECT_EQ(CELIX_ILLEGAL_ARGUMENT, celix_properties_jsonErrorToStatus(json_error_index_out_of_range));
-}
\ No newline at end of file
+}
+
+TEST_F(PropertiesSerializationTestSuite, KeyCollision) {
```

Comment:
Good to see that you found a way to test this corner case.
I though I tried the same, but was struggling to find a way that I could determine the order for 2 hashmap entries.